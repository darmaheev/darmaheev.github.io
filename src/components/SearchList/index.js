import React from "react";
import PropTypes from "prop-types";
import SearchItem from "../SearchItem";
import Pagination from "../Pagination";
import {
  Wrapper,
  List,
  NotFound,
} from "./styles";

const SearchList = ({
  pokemonList,
  page,
  changePage,
  ...restProps
}) => (
  <Wrapper>
    <List>
      {pokemonList && Array.isArray(pokemonList) && pokemonList.map(pokemon => (
        <SearchItem key={pokemon.id} {...restProps} pokemon={pokemon}/>))}

    </List>
    {pokemonList.length > 1 && (
      <Pagination
        total={page.total}
        pageSize={page.pageSize}
        start={page.start}
        onPageClick={changePage}
        withPages
        withPageSize
      />
    )}
    {pokemonList.length === 0 && (
      <NotFound>
        Pokemons not found
      </NotFound>
    )}
  </Wrapper>
);

SearchList.defaultProps = {
  pokemonList: [],
  page: {
    start: 0,
    pageSize: 0,
    total: 0
  }
};

SearchList.propTypes = {
  pokemonList: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number
  })),
  page: PropTypes.shape({
    start: PropTypes.number,
    pageSize: PropTypes.number,
    total: PropTypes.number
  })
};

export default SearchList;
