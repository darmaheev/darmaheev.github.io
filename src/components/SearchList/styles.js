import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const List = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-shrink: 0;
  justify-content: center;
  
`;

export const NotFound = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px;
`;
