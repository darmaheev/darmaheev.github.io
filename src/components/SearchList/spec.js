import React from "react";
import { MemoryRouter } from "react-router-dom";
import SearchList from "./index";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders searchList with default props", () => {
  const pokemonList = [];
  const page = {};
  const changePage = jest.fn();

  const searchList = shallow(
    <SearchList
      pokemonList={pokemonList}
      page={page}
      changePage={changePage}
    />
  );
  expect(searchList).toMatchSnapshot();
});

it("renders searchList with data", () => {
  const pokemonList = [{ id: 1, name: "bulbasaur", sprites: {}}];
  const page = {
    start: 0,
    pageSize: 20,
    total: 1
  };
  const changePage = jest.fn();

  const searchList = mount(
    <MemoryRouter>
      <SearchList
        pokemonList={pokemonList}
        page={page}
        changePage={changePage}
      />
    </MemoryRouter>
  );
  expect(searchList.text()).toContain("bulbasaur");
});

it("without data", () => {
  const pokemonList = [];
  const page = {
    start: 0,
    pageSize: 20,
    total: 0
  };
  const changePage = jest.fn();

  const searchList = mount(
    <MemoryRouter>
      <SearchList
        pokemonList={pokemonList}
        page={page}
        changePage={changePage}
      />
    </MemoryRouter>
  );
  expect(searchList.text()).toContain("Pokemons not found");
});