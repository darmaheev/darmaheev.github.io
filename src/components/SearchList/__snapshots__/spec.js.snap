// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders searchList with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchList
    changePage={[MockFunction]}
    page={Object {}}
    pokemonList={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef />,
        false,
        <ForwardRef>
          Pokemons not found
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-kgoBCf",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-shrink: 0;
  justify-content: center;
  
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-kgoBCf",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Pokemons not found",
        },
        "ref": null,
        "rendered": "Pokemons not found",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-kGXeez",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-kGXeez",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-chPdSV",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-chPdSV",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef />,
          false,
          <ForwardRef>
            Pokemons not found
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-kgoBCf",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-shrink: 0;
  justify-content: center;
  
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-kgoBCf",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Pokemons not found",
          },
          "ref": null,
          "rendered": "Pokemons not found",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-kGXeez",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-kGXeez",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-chPdSV",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-chPdSV",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
